pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -65,8 +65,6 @@ func NewTargetConfigController(
 		c.sync,
 	).ResyncEvery(
 		20*time.Minute, // TODO: figure out what's the idead resync time for this controller
-	).WithSyncDegradedOnError(
-		operatorClient,
```

Comment:
okay, so this is the controller code that was causing `XXXDegraded:false` condition to be added intermittently and hence I have removed it. This controller purely exists to remove older conditions anyways.